src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -342,6 +345,27 @@ public function iShouldBeOnTheCheckoutCompleteStep(): void
         );
     }
 
+    /**
+     * @Then I should not be able to confirm order because products does not fit :shippingMethod requirements
+     */
+    public function iShouldNotBeAbleToConfirmOrderBecauseDoesNotBelongsToShippingCategory(
+        ShippingMethodInterface $shippingMethod
+    ): void {
+        $this->iProceededWithShippingMethod($shippingMethod);
+
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+
+        Assert::true($this->isViolationWithMessageInResponse(
+            $response,
+            sprintf('Product does not fit requirements for %s shipping method. Please reselect your shipping method.',
+                $shippingMethod->getName()
+            )
+        ));
+
+        Assert::same($response->getStatusCode(), 400);
```

Comment:
I would suggest moving this check before the one from line 359 ( comparing two integers has less cost) 🚀